devtools/x/src/lint/guppy.rs

```diff
@@ -507,3 +507,167 @@ impl PackageLinter for OnlyPublishToCratesIo {
         Ok(RunStatus::Executed)
     }
 }
+
+// Ensure that Move crates do not depend on Diem crates.
+#[derive(Debug)]
+pub struct MoveCratesDontDependOnDiemCrates;
+
+// List of existing move to diem dependencies.
+// The goal is eliminate all of them gradually.
+// In general, you should refrain from adding new entries to this list.
+const EXISTING_MOVE_TO_DIEM_DEPS: &[(&str, &str)] = &[
+    ("language-benchmarks", "language-e2e-tests"),
+    ("language-benchmarks", "diem-vm"),
+    ("language-benchmarks", "diem-types"),
+    ("language-benchmarks", "diem-state-view"),
+    ("language-benchmarks", "diem-proptest-helpers"),
+    ("bytecode-verifier", "diem-types"),
+    ("bytecode-verifier-tests", "diem-types"),
+    ("invalid-mutations", "diem-types"),
+    ("invalid-mutations", "diem-proptest-helpers"),
+    ("compiler", "diem-types"),
+    ("bytecode-source-map", "diem-types"),
+    ("ir-to-bytecode", "diem-types"),
+    ("ir-to-bytecode-syntax", "diem-types"),
+    ("compiled-stdlib", "diem-types"),
+    ("compiled-stdlib", "diem-proptest-helpers"),
+    ("compiled-stdlib", "diem-framework"),
+    ("compiled-stdlib", "diem-crypto"),
+    ("move-ir-types", "diem-types"),
+    ("move-lang", "fallible"),
+    ("move-lang", "diem-types"),
+    ("move-lang", "datatest-stable"),
+    ("move-lang-functional-tests", "functional-tests"),
```

Comment:
Aren't "functional-tests" move functional tests?

Reply:
It's a bit complicated. Long story short:
1. The framework (`functional-tests`) itself is more of a Diem crate than a Move one.
2. The tests are mixed -- they contain both Move and Diem tests.

Eventually we'll need to split the framework into a Move one and a Diem one and migrate all the move tests. For now the crate should be marked as a Diem crate.

Reply:
These do appear only used by ir-testsutie and move-lang-functional-tests.

Reply:
If there are a subset of tests that should be diem-functional-tests, then I still think that functional-tests should be under Move, since it is used by ir-testsuite too. In other words, it's a functional test framework for move source/ir testing, that diem can import.

Reply:
> In other words, it's a functional test framework for move source/ir testing, that diem can import.
Unfortunately this is not the case. The existing functional test framework is highly coupled with Diem concepts like transactions, gas, state views etc. The Move framework we'll implement does not need and must not contain any of these -- it will mostly be a fresh rewrite with a more simplified model, borrowing some of the ideas. 

In other words, the existing functional test crate will stay in the Diem repo and should be treated as a Diem thing. With that being said I should probably label "move-lang-functional-tests" as Diem to avoid any confusion here.